lib/ar_lazy_preload/active_record/collection_proxy.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module ArLazyPreload
+  # ActiveRecord::AssociationRelation patch for setting up lazy_preload_values based on
+  # owner context
+  module CollectionProxy
+    def self.included(base)
+      base.prepend(Relation)
+      base.prepend(InstanceMethods)
+    end
+
+    module InstanceMethods
+      private
+
+      def _ar_lazy_preload_records
```

Comment:
I guess it will be enough to define a private method `ar_lazy_preload_records` inside the `CollectionModule` module